pylint/config/arguments_manager.py

```diff
@@ -41,17 +42,17 @@ def __init__(self, prog: str, usage: Optional[str] = None) -> None:
         self._argument_groups_dict: Dict[str, argparse._ArgumentGroup] = {}
         """Dictionary of all the argument groups."""
 
+        self._option_dicts: Dict[str, OptionDict] = {}
+        """All option dictionaries that have been registered."""
+
     def _register_options_provider(self, provider: "_ArgumentsProvider") -> None:
         """Register an options provider and load its defaults."""
         for opt, optdict in provider.options:
+            self._option_dicts[opt] = optdict
             argument = _convert_option_to_argument(opt, optdict)
             section = argument.section or provider.name.capitalize()
 
-            # pylint: disable-next=fixme
-            # TODO: Optparse: Always define option_groups_descs on ArgumentsProvider
-            section_desc = None
-            if hasattr(provider, "option_groups_descs"):
-                section_desc = provider.option_groups_descs.get(section, None)  # type: ignore[attr-defined]
+            section_desc = provider.option_groups_descs.get(section, None)
```

Comment:
get default value is always None for a dict. I should have said that there were other instances of this in the last MR but it's not that important.

Reply:
Ah yeah, force of habit. Want me to remove it here?

Reply:
Let's keep it that way, being explicit can be considered better too and we have other more pressing issues 😄